datus/tools/bi_tools/superset_adaptor.py

```
@@ -0,0 +1,920 @@
+    def _authenticate(self) -> None:
+        if self.auth_type() == AuthType.LOGIN:
+            payload = {
+                "username": self.auth_params.username,
+                "password": self.auth_params.password,
+                "refresh": True,
+                "provider": "db",
+            }
+            try:
+                response = self._request("POST", "security/login", require_auth=False, json=payload)
+            except SupersetAdaptorError as exc:
+                raise SupersetAdaptorError(f"Authentication failed: {exc}") from exc
```

Ruff (0.14.10)

Avoid specifying long messages outside the exception class

(TRY003)